### PR TITLE
Extend Post Execution Class Properties

### PR DESCRIPTION
**Pull request by fa9r**
## Describe changes
I extended the post execution classes to have all config contents directly accessible as properties.

The following properties were added:
- `pipeline.docstring`
- `pipeline.spec`
- `run.settings`
- `run.extra`
- `run.enable_cache`
- `step.settings`
- `step.extra`
- `step.enable_cache`
- `step.step_operator`
- `step.experiment_tracker`
- `step.spec`

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/mlops-stacks/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/post_execution/pipeline_run.py [hunk 1]
```diff
@@ -119,6 +119,36 @@ def pipeline_configuration(self) -> Dict[str, Any]:
...
+
+        Returns:
+            The pipeline settings.
+        """
+        settings = self.pipeline_configuration["settings"]
+        return cast(Dict[str, Any], settings)
+
+    @property
+    def extra(self) -> Dict[str, Any]:
+        """Returns the pipeline extras.
```
[the remark this answers is not in the document]
Comment (htahir1) [reply src/zenml/post_execution/pipeline_run.py:134]: I have an explaination here that might be helpful https://app.gitbook.com/o/-MCl1Hlw9oU4xibZ-ymf/s/IygujOLUuJl3QaGTlW4u/advanced-guide/pipelines/settings

File: src/zenml/post_execution/step.py [hunk 2]
```diff
@@ -134,6 +134,69 @@ def step_configuration(self) -> Dict[str, Any]:
...
+
+        Returns:
+            The step operator of the step.
+        """
+        step_operator = self.step_configuration["config"]["step_operator"]
+        return cast(Optional[str], step_operator)
+
+    @property
+    def experiment_tracker(self) -> Optional[str]:
+        """Returns the experiment tracker of the step.
```
Comment (htahir1) [inline src/zenml/post_execution/step.py:179]: Returns the `name of the experiment tracker`

**Review by htahir1 — COMMENTED** [review]
Two comments otherwise LGTM!

File: src/zenml/post_execution/step.py (hunk 2 above)
Comment (fa9r) [reply src/zenml/post_execution/step.py:179]: Good catch, done!

**Review by htahir1 — APPROVED** [review]

**Pull request merged**